html/janus.js

```diff
@@ -2206,7 +2210,7 @@ function Janus(gatewayCallbacks) {
 			}
 		}
 		// If we're updating, check if we need to remove/replace one of the tracks
-		if(media.update && !config.streamExternal) {
+		if(media.update && !config.streamExternal || (config.streamExternal && (media.replaceAudio || media.replaceVideo))) {
```

Comment:
I think the check as you modified it is ambiguous, since there's an `&&` and an `||` in the same "sentence", so it's not clear what the AND applies to, and what the OR either: did you mean to group the old check in parenthesis as well, and make it an OR between the two groups?

Reply:
It is ambiguous, I will make it more clear with an extra set of parenthesis:

`if(media.update && (!config.streamExternal || (config.streamExternal && (media.replaceAudio || media.replaceVideo))))`